packages/react/context-menu/src/ContextMenu.tsx

```diff
@@ -14,14 +14,13 @@ import type { Point, MeasurableElement } from '@radix-ui/utils';
 const CONTEXT_MENU_NAME = 'ContextMenu';
 
 type ContextMenuContextValue = {
-  open: boolean;
-  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
   anchorPointRef: React.MutableRefObject<Point>;
   anchorRef: React.MutableRefObject<MeasurableElement | null>;
+  open: boolean;
+  onOpenChange(open: boolean): void;
```

Comment:
Again, not sure how I feel here, even the change of signature, we're putting a setter in it, what if I actually want to use the `prevState => newState` syntax? I'm kinda confused why we're changing all these… I'm probably not seeing something.

Reply:
I've explained why I renamed them [here]( https://github.com/radix-ui/primitives/pull/479#discussion_r582808283) but the reason for changing the signature was because we don't use the one you're mentioning so if we don't need it, I thought why complicate it? My thinking was that **when** we want that callback signature, we would update it then. It was one of those "only introduce complexity when inevitable" things for me.

I've actually always disliked the way we do `context.setOpen(prevOpen => !prevOpen)` and instead would have prefer nsomething like this:

```jsx
<ContextProvider onOpenToggle={() => setOpen(prevOpen => !prevOpen)} />
```

That way we're not spreading implementation detail across all the parts.